boards/nxp/s32z2xxdc2/s32z2xxdc2_s32z270_rtu0_B.overlay

```diff
@@ -16,3 +16,19 @@
 	pinctrl-names = "default";
 	status = "okay";
 };
+
+/* Connect J32 pin 5 <-> J32 pin 3, J32 pin 6 <-> J32 pin 4 */
```

Comment:
So the EEPROM is present on the board, but not connected by default? If so, the devicetree node should be disabled by default.

Reply:
yes, it didn't connect by default, I enabled it in test instead

Reply:
I updated it

Reply:
This EEPROM is meant to hold configuration used by the boot ROM, and the jumper `J32` seems to be used to select whether this is taken from the motherboard or the daughterboard (this board). By making this EEPROM available at board's overlay, you are encouraging users to give a different use than intended, just to enable I2C testing through the EEPROM interface. I'm not sure this is the right approach..

Reply:
I fully agree. The EEPROM tests, which are enabled, are destructive. They will alter the content of the EEPROM.

Reply:
I updated and enabled it in the test

Reply:
> I updated and enabled it in the test

The problem still persists no matter where you define the eeprom node, whether in the board or tests overlay. I don't think using this onboard EEPROM for testing I2C is correct, and IMO you should enable I2C API tests instead.

Reply:
Hi I remove enable EEPROM test, Testing for I2C will be manually tested by using _**tests\drivers\i2c\i2c_target_api**_ with a mother board connected but it won't be introduced, just keep internal testing